tracer/src/Datadog.Trace/Ci/CiEnvironment/GitCommandGitInfoProvider.cs

```diff
@@ -0,0 +1,144 @@
+            }
+            else
+            {
+                localGitInfo.Errors.Add($"Error getting repository URL: {repositoryOutput?.Error}");
+            }
+
+            // Get the branch name
+            var branchOutput = ProcessHelpers.RunCommand(
+                new ProcessHelpers.Command(
+                    cmd: "git",
+                    arguments: "rev-parse --abbrev-ref HEAD",
+                    workingDirectory: gitDirectory.FullName,
+                    useWhereIsIfFileNotFound: true));
+            if (branchOutput?.ExitCode == 0 && branchOutput.Output.Trim() is { Length: > 0 } branchName && branchName != "HEAD")
+            {
+                localGitInfo.Branch = branchName;
+            }
+            else
+            {
+                localGitInfo.Errors.Add($"Error getting branch name: {branchOutput?.Error}");
+            }
+
+            // Get the remaining data from the log -1
+            var gitLogOutput = ProcessHelpers.RunCommand(
+                new ProcessHelpers.Command(
+                    cmd: "git",
+                    arguments: """log -1 --pretty='%H|,|%at|,|%an|,|%ae|,|%ct|,|%cn|,|%ce|,|%B'""",
+                    workingDirectory: gitDirectory.FullName,
+                    useWhereIsIfFileNotFound: true));
+            if (gitLogOutput?.ExitCode != 0)
+            {
+                localGitInfo.Errors.Add($"Error getting git log: {gitLogOutput?.Error}");
+                return false;
+            }
+
+            var gitLogDataArray = gitLogOutput.Output.Trim().Split(["|,|"], StringSplitOptions.None);
+            if (gitLogDataArray.Length < 8)
+            {
+                localGitInfo.Errors.Add($"Git log output does not contain the expected number of fields: {gitLogOutput.Output}");
+                return false;
+            }
+
+            // Parse author and committer dates from Unix timestamp
+            if (!long.TryParse(gitLogDataArray[1], out var authorUnixDate))
+            {
+                localGitInfo.Errors.Add("Error parsing author date from git log output");
+                return false;
+            }
+
+            if (!long.TryParse(gitLogDataArray[4], out var committerUnixDate))
+            {
+                localGitInfo.Errors.Add("Error parsing committer date from git log output");
+                return false;
+            }
+
+            // Populate the localGitData struct with the parsed information
+            localGitInfo.Commit = gitLogDataArray[0];
+            localGitInfo.AuthorDate = UnixTimeStampToDateTime(authorUnixDate);
+            localGitInfo.AuthorName = gitLogDataArray[2];
+            localGitInfo.AuthorEmail = gitLogDataArray[3];
+            localGitInfo.CommitterDate = UnixTimeStampToDateTime(committerUnixDate);
+            localGitInfo.CommitterName = gitLogDataArray[5];
+            localGitInfo.CommitterEmail = gitLogDataArray[6];
+            localGitInfo.Message = string.Join("|,|", gitLogDataArray.Skip(7)).Trim();
```

Comment:
LINQ? Where's the real Tony? What have you done with him?! 🤨

Reply:
yeah, I know right? well is not in a hot path 🤷🏻 only runs once... but.. yeah, shame on me! :shame: